Drv/LinuxUartDriver/LinuxUartDriver.cpp

```diff
@@ -0,0 +1,404 @@
+            return false;
+        }
+    }
+
+    NATIVE_INT_TYPE relayRate = B0;
+    switch (baud) {
+        // TODO add more as needed
+        case BAUD_9600:
+            relayRate = B9600;
+            break;
+        case BAUD_19200:
+            relayRate = B19200;
+            break;
+        case BAUD_38400:
+            relayRate = B38400;
+            break;
+        case BAUD_57600:
+            relayRate = B57600;
+            break;
+        case BAUD_115K:
+            relayRate = B115200;
+            break;
+        case BAUD_230K:
+            relayRate = B230400;
+            break;
+#if defined TGT_OS_TYPE_LINUX
+        case BAUD_460K:
+            relayRate = B460800;
+            break;
+        case BAUD_921K:
+            relayRate = B921600;
+            break;
+#endif
+        default:
+            FW_ASSERT(0, baud);
+            break;
+    }
+
+    struct termios newtio;
+
+    stat = tcgetattr(fd, &newtio);
+    if (-1 == stat) {
+        DEBUG_PRINT("tcgetattr UART fd %d failed\n", fd);
+        close(fd);
+        Fw::LogStringArg _arg = device;
+        Fw::LogStringArg _err = strerror(errno);
+        this->log_WARNING_HI_DR_OpenError(_arg, fd, _err);
+        return false;
+    }
+
+    // CS8 = 8 data bits, CLOCAL = Local line, CREAD = Enable Receiver
+    // TODO PARENB for parity bit
+    /*
+      Even parity (7E1):
+      options.c_cflag |= PARENB
+      options.c_cflag &= ~PARODD
+      options.c_cflag &= ~CSTOPB
+      options.c_cflag &= ~CSIZE;
+      options.c_cflag |= CS7;
+      Odd parity (7O1):
+      options.c_cflag |= PARENB
+      options.c_cflag |= PARODD
+      options.c_cflag &= ~CSTOPB
+      options.c_cflag &= ~CSIZE;
+      options.c_cflag |= CS7;
+     */
+    newtio.c_cflag = CS8 | CLOCAL | CREAD;
+
+    switch (parity) {
+        case PARITY_ODD:
+            newtio.c_cflag |= (PARENB | PARODD);
+            break;
+        case PARITY_EVEN:
+            newtio.c_cflag |= PARENB;
+            break;
+        case PARITY_NONE:
+            newtio.c_cflag &= ~PARENB;
+            break;
+        default:
+            FW_ASSERT(0, parity);
+            break;
+    }
+
+    // Set baud rate:
+    stat = cfsetispeed(&newtio, relayRate);
+    if (stat) {
+        DEBUG_PRINT("cfsetispeed failed\n");
+        close(fd);
+        Fw::LogStringArg _arg = device;
+        Fw::LogStringArg _err = strerror(errno);
+        this->log_WARNING_HI_DR_OpenError(_arg, fd, _err);
+        return false;
+    }
+    stat = cfsetospeed(&newtio, relayRate);
+    if (stat) {
+        DEBUG_PRINT("cfsetospeed failed\n");
+        close(fd);
+        Fw::LogStringArg _arg = device;
+        Fw::LogStringArg _err = strerror(errno);
+        this->log_WARNING_HI_DR_OpenError(_arg, fd, _err);
+        return false;
+    }
+
+    // Raw output:
+    newtio.c_oflag = 0;
+
+    // set input mode (non-canonical, no echo,...)
+    newtio.c_lflag = 0;
+
+    // TODO if parity, then do input parity too
```

Comment:
Is this still a TODO?

Reply:
I have no idea, I looked at the code and it looked like that was completed....so I removed the todo.

Reply:
Thanks. But I still see the TODO comment.

Reply:
Could you remove the TODO comment?

Reply:
Done